migration/incremental/015-tracing-compression.sql

```diff
@@ -190,34 +190,197 @@ GRANT USAGE ON TYPE ps_trace.trace_id TO prom_reader;
 -- ps_trace.trace_id and uuid are binary coercible
 CREATE CAST(ps_trace.trace_id as uuid) WITHOUT FUNCTION AS IMPLICIT;
 CREATE CAST(uuid AS ps_trace.trace_id) WITHOUT FUNCTION AS IMPLICIT;
+/* Move the old tag_map type out of the way */
+ALTER DOMAIN ps_trace.tag_map
+    RENAME TO tag_map_old;
+ALTER DOMAIN ps_trace.tag_map_old
+    SET SCHEMA _ps_trace;
+
+/* Define custom tag_map and tag_v json aliases
+    * NOTE: We need to keep an eye on upstream jsonb type
+    * and keep these up-to-date.
+    */
+
+    CREATE TYPE ps_trace.tag_map;
+    CREATE TYPE _ps_trace.tag_v;
+
+    CREATE OR REPLACE FUNCTION ps_trace.tag_map_in(cstring)
```

Comment:
all these functions should also be defined in an idempotent script because all our functions should appear in an idempotent script